src/main/kotlin/org/openrewrite/kotlin/internal/PsiElementAssociations.kt

```diff
@@ -197,6 +201,7 @@ class PsiElementAssociations(val typeMapping: KotlinTypeMapping, val file: FirFi
         val fir = primary(psi) ?: return null
         return when (fir) {
             is FirResolvedQualifier -> ExpressionType.QUALIFIER
+            is FirArrayOfCall -> ExpressionType.METHOD_INVOCATION
```

Comment:
I'll look more into this tomorrow.

`FirArrayOfCall` implements `FirCall` and is not a `FirFunctionCall`. So, from the `FIR` it isn't a `METHOD_INVOCATION`.

There is however, an `arrayOf` function in `Library.kt`. But `Library.kt` isn't resolved until the IR phase, so we may not be able to get the functions type.

`FirArrayOfCall` does contain a `typeRef` since it is an expression and will return the parameterized type from the functions cast: `public inline fun <T> arrayOf(vararg elements: T): Array<T> = elements as Array<T>`. So, using type(firArrayOfCall) will work, but a method invocation won't.